k8s/upgrade/src/events/event_recorder.rs

```diff
@@ -170,21 +177,29 @@ impl EventRecorderBuilder {
             event_loop_handle,
             source_version,
             target_version,
+            product_name: self.product_name,
         })
     }
 }
 
 /// This is a wrapper around a kube::runtime::events::Recorder.
-pub struct EventRecorder {
+pub struct EventRecorder<'a> {
```

Comment:
Are all the other fields already "settable" ?

Reply:
No, the source-version and the target-version are settable. This is essentially due to the API not being designed well. The rest of the members' values are available at startup, so they are populated via the builder. The source-version and the target-version are picked up at run-time, so they are populated later.